# -*- coding: utf-8 -*-
#!/usr/bin/env python
PACKAGE = "recipelab"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 파라미터(변수명:x_kp, 변수타입:double_t, 레벨:0, 설명:[gain]x_Kp, 기본값, 하한값, 상한값)
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
gen.add("x_Kp", double_t, 0, "[gain]x_Kp", 0, -100, +100)
gen.add("x_Ki", double_t, 0, "[gain]x_Ki", 0, -100, +100)
gen.add("x_Kd", double_t, 0, "[gain]x_Kd", 0, -100, +100)
gen.add("x_Kc", double_t, 0, "[gain]x_Kc", 0, -100, +100)

gen.add("y_Kp", double_t, 0, "[gain]y_Kp", 0, -100, +100)
gen.add("y_Ki", double_t, 0, "[gain]y_Ki", 0, -100, +100)
gen.add("y_Kd", double_t, 0, "[gain]y_Kd", 0, -100, +100)
gen.add("y_Kc", double_t, 0, "[gain]y_Kc", 0, -100, +100)

gen.add("rz_Kp", double_t, 0, "[gain]rz_Kp", 0, -100, +100)
gen.add("rz_Ki", double_t, 0, "[gain]rz_Ki", 0, -100, +100)
gen.add("rz_Kd", double_t, 0, "[gain]rz_Kd", 0, -100, +100)
gen.add("rz_Kc", double_t, 0, "[gain]rz_Kc", 0, -100, +100)

# 파라미터(패키지명, 노드명, 설정파라미터접두사(ex)<name>Config.h, <name>Config.py)
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
exit(gen.generate(PACKAGE, "gain_tuning", "GainTuning"))